### ISSUE TITLE
Password Decode Failure, Step 3 Installation (Access denied for user 'username'@'hostname' (using password: YES))

### ISSUE THREAD

**Issue by JonathanCJames** [opencart/opencart#3185]
When installing version 2.0.3.1 on localhost (using Linux SUSE), I was given the error:
"Access denied for user 'root'@'localhost' (using password: YES)"

The password I always use for accessing 127.0.0.1 contains an ampersand; this was not being decoded properly.

You have already deployed 'html_entity_decode' for the file 'install/controller/step_3.php' on lines 35 and 73.  However to resolve the issue, I also had to add 'html_entity_decode' to line 267 as follows:
Replaced:  $this->request->post['db_password']
With:  html_entity_decode($this->request->post['db_password'], ENT_QUOTES, 'UTF-8')

I also needed to add decoding for the file 'instal/model/install.php' on line 4 as follows:
Replaced:  $data['db_password']
With:  html_entity_decode($data['db_password'])

Decoding the password in these two places resolved the issue.

**Comment by JonathanCJames** [comment]
Same problem exists for admin password.  Don't currently have time to fix this, too.

**Comment by tflight** [comment]
Just pinging this is still an issue in 2.2.0.0.

**Comment by drkskwlkr** [comment]
@tflight: the fix seems to be implemented but today I found an edge case where special characters in DB password are still an issue even after using html_entity_decode (I tested on 2.1.0.2).

The password that caused me trouble contained a `%&` sequence in the middle and step 3 of the OpenCart installation failed on a blank page with MySQL 'access denied' message. Once I replaced the `%&` with `##` the problem went away.
